Source/WebCore/Modules/mediastream/ImageCapture.cpp

```diff
@@ -108,19 +109,20 @@ void ImageCapture::getPhotoCapabilities(DOMPromiseDeferred<IDLDictionary<PhotoCa
         return;
     }
 
-    m_track->getPhotoCapabilities()->whenSettled(RunLoop::protectedMain(), [this, protectedThis = Ref { *this }, promise = WTFMove(promise), identifier = WTFMove(identifier)] (auto&& result) mutable {
-        legacyQueueTaskKeepingObjectAlive(*this, TaskSource::ImageCapture, [this, promise = WTFMove(promise), result = WTFMove(result), identifier = WTFMove(identifier)] () mutable {
+    Ref protectedThis = Ref { *this };
+    m_track->getPhotoCapabilities()->whenSettled(RunLoop::protectedMain(), [protectedThis = WTFMove(protectedThis), promise = WTFMove(promise), identifier = WTFMove(identifier)] (auto&& result) mutable {
```

Comment:
doing `protectedThis = Ref { *this }` looks simpler to me that creating the object first and moving it in the capture

Reply:
I totally agree with Jean Yves here and the same issue applies elsewhere in the PR. We don't do local variable protectors outside lambda captures anymore.